### PR TITLE
JIRA HS-1 Student Browse Credit Page

### PR DESCRIPTION
**Pull request by mitrasingh**
This pull request implements the **Student Browse Credits Page**. 

Changes include:
- new UI for students/page.tsx
- "dummy" json data for credits title/images 
- credit list component to map json data into page
- components folder created for application components 
- api folder created for json fetch using local server (to create local server "npx json-server --watch src/app/students/credits.json --port 8000" in terminal)

Notes:
- this pull also includes Feedback modal which is not completed but would be considered another sprint
- components/Modal folder will have other modals such as log out and confirmations including feedback so I left that folder to be a general name
- creditlist.tsx and credit.json is currently only associated with the students page so I kept it within that folder

### CODE REVIEW

File: src/app/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+type ButtonProps = {
+	text: string;
+	width: string;
+};
+
+const Button = ({ text, width }: ButtonProps) => {
+	return (
+		<button
+			className={`h-[39px] w-[${width}] rounded-full bg-[#805DBE] text-white text-xs font-bold hover:bg-violet-700`}
```
Comment (thedomcab) [inline src/app/components/Button/Button.tsx:9]: don't use height or width use padding. you should never use height or width.

File: src/app/globals.css [hunk 2]
```diff
@@ -1,35 +1,61 @@
...
 }
 
-main{
-  flex:1;
-  background-color: #ffffff;
-  border-radius: 20px;
-  margin: 0 50px;
-}
\ No newline at end of file
+/* Track */
```
Comment (thedomcab) [inline src/app/globals.css:46]: Why was this added?

Comment (mitrasingh) [reply src/app/globals.css]: They are for the inner scroll within the browse credits page. I went ahead and commented and added the additional styling being used in our current main branch.

File: src/app/components/Button/Button.tsx (hunk 1 above)
Comment (mitrasingh) [reply src/app/components/Button/Button.tsx]: Apologies on that. I removed all props, made it a general button and moved this component within the route folder.

**Comment by mitrasingh** [comment]
@thedomcab resolved all conflicts

**Comment by mitrasingh** [comment]
Resolved global css conflict, ready for review.

File: src/app/components/Modal/FeedbackModal.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+		if (isQuestionOneDropdown) {
+			setIsQuestionOneDropDown(false);
+		}
+	};
+
+	if (!isVisible) return null;
+
+	return (
+		<div className="fixed inset-0 bg-black bg-opacity-50 backdrop-blur-sm flex justify-center items-center">
+			<div className="w-[55%] h-[61%] flex flex-col p-8 rounded-lg shadow bg-[#f9f7fc]">
```
Comment (thedomcab) [inline src/app/components/Modal/FeedbackModal.tsx:32]: we should never set height unless it is a small item. Could you use padding plaease?

File: src/app/components/Modal/FeedbackModal.tsx [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+						{pageTitle}
+					</h2>
+
+					{/* Question One*/}
+					<section>
+						<h2 className="font-medium mt-6 mb-2 text-gray-800">
+							Is this page easy to use?
+						</h2>
+						<div
+							onClick={toggleQuestionOneDropdown}
```
Comment (thedomcab) [inline src/app/components/Modal/FeedbackModal.tsx:50]: Use a button for on click please

File: src/app/components/Modal/FeedbackModal.tsx [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+							className="w-36 h-8 p-2 flex items-center justify-between rounded drop-shadow-sm bg-white hover:drop-shadow"
+						>
+							Select
+							<svg
+								xmlns="http://www.w3.org/2000/svg"
+								fill="none"
+								viewBox="0 0 24 24"
+								strokeWidth={2}
+								stroke="currentColor"
```
Comment (thedomcab) [inline src/app/components/Modal/FeedbackModal.tsx:59]: update using css for html attributes

File: src/app/components/Modal/FeedbackModal.tsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+								</li>
+							</ul>
+						)}
+
+						{/* Question Two*/}
+						<h2 className="font-medium mt-6 mb-2 text-gray-800">
+							How would you rate the design of this page?
+						</h2>
+						<div
+							onClick={toggleQuestionTwoDropdown}
```
Comment (thedomcab) [inline src/app/components/Modal/FeedbackModal.tsx:94]: use a button for this on click event

File: src/app/students/browsecredits/CreditList.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+							<div key={credit.id}>
+								<figure className="h-[12.5rem] w-[12.5rem] md:h-[7.8125rem] md:w-[7.8125rem] lg:h-[12.5rem] lg:w-[12.5rem] ">
+									<Link href="/">
+										<Image
+											src={credit.url}
+											width={0}
+											height={0}
+											sizes="100vw"
+											alt="Picture of the credit"
+											className="w-full h-full object-cover transition-opacity opacity-0 duration-300 border-2 border-transparent hover:border-[#805DBE]"
```
Comment (thedomcab) [inline src/app/students/browsecredits/CreditList.tsx:56]: don't need h-full for image. this is automatic

File: src/app/students/browsecredits/Search.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+			<figure>
+				<svg
+					className="-ml-10"
+					role="graphic-content"
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 50 50"
+					fill="#9ca3af"
+					width="24px"
+					height="24px"
+				>
```
Comment (thedomcab) [inline src/app/students/browsecredits/Search.tsx:27]: use css for html attributs

File: src/app/components/Modal/FeedbackModal.tsx (hunk 3 above)
Comment (mitrasingh) [reply src/app/components/Modal/FeedbackModal.tsx]: resolved

File: src/app/components/Modal/FeedbackModal.tsx (hunk 4 above)
Comment (mitrasingh) [reply src/app/components/Modal/FeedbackModal.tsx:50]: resolved, FYI: Feedback Modal UI is not 100 percent, most likely will change later on.

File: src/app/components/Modal/FeedbackModal.tsx (hunk 6 above)
Comment (mitrasingh) [reply src/app/components/Modal/FeedbackModal.tsx:94]: resolved

**Comment by mitrasingh** [comment]
@thedomcab  all conflicts have been resolved, ready for merge

**Review by thedomcab — COMMENTED** [review]
could you remove local server setup to get data?

**Comment by mitrasingh** [comment]
@thedomcab local server setup to fetch data is now removed.

File: src/app/students/browsecredits/BackButton.tsx [hunk 9]
```diff
@@ -0,0 +1,11 @@
+const Button = () => {
+	return (
+		<button
+			className={`h-[2.4375rem] w-[5.125rem] rounded-full bg-[#805DBE] text-white text-xs font-bold hover:bg-violet-700`}
```
Comment (thedomcab) [inline src/app/students/browsecredits/BackButton.tsx:4]: always using padding for buttons.  could you update this?

File: src/app/students/browsecredits/CreditList.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+		<>
+			{Credits && (
+				<>
+					{Credits.filter((credit) => {
+						return search.toLowerCase() === ""
+							? credit
+							: credit.creditname.toLowerCase().includes(search);
+					}).map((credit) => (
+						<div key={credit.id}>
+							<figure className="h-[12.5rem] w-[12.5rem] md:h-[7.8125rem] md:w-[7.8125rem] lg:h-[12.5rem] lg:w-[12.5rem] ">
```
Comment (thedomcab) [inline src/app/students/browsecredits/CreditList.tsx:32]: why is the set width and height? what is this for?

File: src/app/students/browsecredits/credits.json [hunk 11]
```diff
@@ -0,0 +1,62 @@
+[
```
Comment (thedomcab) [inline src/app/students/browsecredits/credits.json:1]: where is this be pulled into? what component?

File: src/app/students/browsecredits/BackButton.tsx (hunk 9 above)
Comment (mitrasingh) [reply src/app/students/browsecredits/BackButton.tsx]: updated to padding, included Link component.

File: src/app/students/browsecredits/CreditList.tsx (hunk 10 above)
Comment (mitrasingh) [reply src/app/students/browsecredits/CreditList.tsx]: Thanks for catching this, removed the width and height.

File: src/app/students/browsecredits/credits.json (hunk 11 above)
Comment (mitrasingh) [reply src/app/students/browsecredits/credits.json]: Removed this json file to avoid confusion. json data (dummy data) now located within CreditList component. This is now resolved. Thanks!

**Comment by mitrasingh** [comment]
@thedomcab PR is ready for review/merge. Keep me posted. Thanks!

**Review by thedomcab — APPROVED** [review]
